File: linked-list/2.3.js
const Node = require('./node');
const n = new Node('a');
n.next = new Node('b');
n.next.next = new Node('c');
n.next.next.next = new Node('d');
n.next.next.next.next = new Node('e');
n.next.next.next.next.next = new Node('f');

function deleteNode(n) {
  if (n === null || n.next === null) {
    return false;
  }
  n.next = n.next.next;
  return true;
}
function printNodes(_node) {
  while (_node !== null) {
    console.log(_node.data);
    _node = _node.next;
  }
}
console.log(deleteNode(n.next.next));
printNodes(n);
console.log(deleteNode(n.next.next.next));
printNodes(n);

File: chapter 2 - linked list/2.3.py
from node import Node

def print_nodes(n):
    while(n != None):
        print(n.data)
        n = n.next

def delete_middle_node(n):
    # TODO: last node case 
    if n == None or n.next == None:
        return False
    else:
        n.next = n.next.next
    return True                
 
if __name__ == "__main__":
    n = Node('a')
    n.next =  Node('b')
    n.next.next =  Node('c') 
    n.next.next.next =  Node('d')
    n.next.next.next.next =  Node('e')
    n.next.next.next.next.next =  Node('f') 
    
    delete_middle_node(n.next.next)
    print_nodes(n)

